File: src/javascript/rectangulo.mjs
import readline from 'node:readline/promises';
const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
async function pedirPositivo(msg){
  while(true){
    const v = Number(await rl.question(msg));
    if(!Number.isNaN(v) && v > 0) return v;
    console.log('Introduce un número positivo');
  }
}
const base = await pedirPositivo('Base: ');
const altura = await pedirPositivo('Altura: ');
console.log('Área =', base*altura);
rl.close();

File: src/c/rectangulo.c
#include <stdio.h>
int main(){
  double base=0, altura=0;
  do{ printf("Base: "); scanf("%lf", &base);}while(base <= 0);
  do{ printf("Altura: "); scanf("%lf", &altura);}while(altura <= 0);
  printf("Área = %f\n", base*altura);
  return 0;
}
